### PR TITLE
Remove concept of RUN_WITH_SUBMIT

### PR DESCRIPTION
**Pull request by jgfouca**
It added a good deal of complexity and broke some usecases just for
the sake of protecting the user from themself. Instead, make batch
script files hidden from user so that they don't need to be
protected in the first place.

Test suite: scripts_regression_tests (on skybridge, batch machine)
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes [CIME Github issue #]

User interface changes?: Y, renaming of batch scripts

Update gh-pages html (Y/N)?:N

Code review: @jedwards4b @billsacks

### CODE REVIEW

**Comment by jgfouca** [comment]
@billsacks , the comments in the st_archive template file say this:

```
template to create a case short term archiving script. This should only ever be called                                                                                                                                                                                             
by case.submit when on batch system. This script only exists as a way of providing                                                                                                                                                                                                 
batch directives. Use case.submit from the command line to run your case. 
```

So, that indicated to me that it should not be called. Perhaps it's a copy/paste error?

As to 2, yes, that's correct. The only UI change is file names of stuff they should have been touching anyway.

File: scripts/lib/CIME/case_setup.py [hunk 1]
```diff
@@ -83,21 +83,18 @@ def _case_setup_impl(case, caseroot, clean=False, test_mode=False, reset=False):
...
         if not test_mode:
             # rebuild the models (even on restart)
             case.set_value("BUILD_COMPLETE", False)
 
             # backup and then clean test script
-            if os.path.exists("case.test"):
-                os.remove("case.test")
-                logger.info("Successfully cleaned test script case.test")
-
-            if os.path.exists("case.testdriver"):
```
Comment (jgfouca) [inline scripts/lib/CIME/case_setup.py]: case.testdriver is ancient legacy stuff, not used anymore.

**Comment by jedwards4b** [comment]
@billsacks you can run just the st_archive script through case.submit like
case.submit --job case.st_archive, you can also do this without submiting
case.submit --job case.st_archve --no-batch

**Comment by billsacks** [comment]
I spoke with @bertinia and she said that there has been the expectation that you should be able to run `./case.st_archive` directly (i.e., the comment you pointed to, @jgfouca , was indeed probably a copy/paste error). @jedwards4b - I see your point about this alternative way to run it, but that sounds a lot harder and less intuitive for users than just running `./case.st_archive`.

So my feeling is that you should revert `case.st_archive` to its old name (without a leading `.`) - just hiding `case.run` and `case.test`.

**Comment by jgfouca** [comment]
@billsacks , the only problem with that approach is that we would then be inconsistent in our naming of batch scripts. I'm leaning towards @jedwards4b and for making case.submit the one entry-point to running anything.

**Comment by jgfouca** [comment]
An alternative would be to provide wrapper script, case.st_archive, which just calls case.submit --job case.st_archive.

**Comment by jedwards4b** [comment]
We could add a short case.st_archive script that just does this:
case.submit --job case.st_archive --no-batch

**Comment by jgfouca** [comment]
@jedwards4b , on it, i see we're on the same wavelength here.

**Comment by jedwards4b** [comment]
hmm - does renaming these cause the --job argument to case.submit to break - @jgfouca did you happen to test this?

**Comment by jgfouca** [comment]
@jedwards4b , I ran the full suite on a batch system. I think I caught all the spots where there was an assumption that the job name matched the script name.

**Comment by billsacks** [comment]
@jgfouca @jedwards4b I'm happy with the approach you mention:

> We could add a short case.st_archive script that just does this:
case.submit --job case.st_archive --no-batch

**Comment by jgfouca** [comment]
OK, I've added the wrapper script we talked about and tested it by-hand.

**Comment by rljacob** [comment]
case.st_archive is run by hand all the time.

**Comment by jgfouca** [comment]
@rljacob , that's OK; it's still supported.

**Pull request merged**